src/main/java/io/lettuce/core/StatefulRedisConnectionImpl.java

```diff
@@ -293,4 +295,29 @@ public ConnectionState getConnectionState() {
         return state;
     }
 
+    public void reauthenticate()
+    {
+        RedisCredentialsProvider credentialsProvider = state.getCredentialsProvider();
+
+        if (credentialsProvider instanceof RedisCredentialsProvider.ImmediateRedisCredentialsProvider) {
```

Comment:
why is this required? Can't the immediate provider return the correct credentialsin `resolveCredentials`?

Reply:
Keep this behavior from other places in the code, see RedisHandshake